### DISCUSSION TITLE
Welcome to Magpie Discussions!

### DISCUSSION THREAD

**Discussion by chu11** [llnl/magpie#329] [Announcements]
Feel free ask questions you’re wondering about or share ideas that don't quite qualify as "issues".  Please submit bugs or feature requests to issues instead.

**Comment by pripiat1986** [comment]
Can Magpie generate configurations for Hadoop and Spark to support RDMA transport?

**Comment by chu11** [comment]
No.  Unless there's been recent updates I'm unaware of, standard Hadoop and Spark do not support RDMA.  The only RDMA support I'm aware of comes from experimental builds coming from Mellanox or Ohio State.

I used to support the Mellanox UDA plugin (Magpie version 1), but abandoned it as that project became stale and users did not seem to care for it.

I experimented with the Ohio State build, but gave up on it as it was clearly "experimental" and I hit too many problems to continue investigating it.

**Reply by pripiat1986** [reply]
Thanks!¡!

**Reply by pripiat1986** [reply]
Hello again, and Magpie can generate configurations for Hadoop and Spark to support OpenMPI ? Does it make sense to use OpenMPI with Spark and Hadoop ? Thanks again.

**Reply by chu11** [reply]
I assume you're asking if if you can launch MPI programs under Hadoop's Yarn scheduler?  I believe some people out there have written plugins to allow this (MPI in general, I cannot speak to OpenMPI specifically), although I have never personally experimented with them, so they are not supported under Magpie.

**Reply by pripiat1986** [reply]
Hello, yes, I know of the existence of interfaces or plugins so that Spark and Hadoop can interact with MPI, my question is if they were supported by Magpie. Thanks again !!!

**Reply by pripiat1986** [reply]
Thanks again !!!

**Reply by chu11** [reply]
Can you point me to the plugins you are experimenting with?  Just so I know for the future.  Thanks.

**Comment by pripiat1986** [comment]
Hello, of course.  For exaple "Alchemist" that allow Spark to interact with the MPI libraries installed in the cluster. 
https://databricks.com/session/alchemist-an-apache-spark-mpi-interface
For Haddop, I am tutoring a doctoral thesis to a student for development of an MPI-IO interface for HDFS and its use in applications
Map-Reduce in MPI.  Thanks !!!

**Reply by pripiat1986** [reply]
Hello one question, do you have any material to make a magpie presentation? thanks.

**Reply by chu11** [reply]
nothing that i can share, but  the overview material was basically identical to what is in Magpie's README

**Reply by pripiat1986** [reply]
Understood, I have to do a presentation at the University and I wanted to know if there was any material that I could base myself on ...
